document/JS.bs

```diff
@@ -0,0 +1,921 @@
+interface Module {
+  static sequence&lt;ModuleExportDescriptor> exports(Module module);
+  static sequence&lt;ModuleImportDescriptor> imports(Module module);
+  static sequence&lt;ArrayBuffer> customSections(Module module, DOMString sectionName);
+};
+</pre>
+
+<!-- Any use for this?
+<div algorithm="name of the export">
+    The <dfn>name of the export</dfn> |export| in the module |module| is defined as follows:
+    1. If |export| is of the form [=𝖿𝗎𝗇𝖼=] |funcidx| and |module| has a [=name section=] with a [=function names subsection=],
+        1. Within <code>[=funcnamesubsec=]</code>, search for an entry <code>[=nameassoc=]</code> of the form |funcidx| <code>name</code>. If such an entry is found, return <code>name</code> [=UTF-8 decoder|decoded as a UTF-8=] into a JavaScript String.
+    1. Return the empty String.
+</div>
+-->
+
+<div algorithm>
+  The <dfn>string value of the extern type</dfn> |type| is
+    * "function" if |type| is of the form 𝖿𝗎𝗇𝖼 functype
+    * "table" if |type| is of the form 𝗍𝖺𝖻𝗅𝖾 tabletype
+    * "memory" if |type| is of the form 𝗆𝖾𝗆    memtype
+    * "global" if |type| is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 globaltype
+</div>
+
+<div algorithm>
+    The <dfn method for="Module">exports(moduleObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |module| be |moduleObject|.\[[Module]].
+    1. Let |exports| be an empty [=list=].
+    1. For each (|name|, |type|) in [=module_exports=](|module|)
+        1. Let |kind| be the [=string value of the extern type=] |type|.
+        1. Let |obj| be a new {{ModuleExportDescriptor}} dictionary with {{ModuleExportDescriptor/name}} |name| and {{ModuleExportDescriptor/kind}} |kind|.
+        1. [=Append=] |obj| to the end of |exports|.
+    1. Return |exports|.
+</div>
+
+<div algorithm>
+    The <dfn method for="Module">imports(moduleObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |module| be |moduleObject|.\[[Module]].
+    1. Let |imports| be an empty [=list=].
+    1. For each (|module|, |name|, |type|) in [=module_imports=](|module|),
+        1. Let |kind| be the [=string value of the extern type=] |type|.
+        1. Let |obj| be a new {{ModuleImportDescriptor}} dictionary with {{ModuleImportDescriptor/module}} |module|, {{ModuleImportDescriptor/name}} |name| and {{ModuleImportDescriptor/kind}} |kind|.
+        1. [=Append=] |obj| to the end of |imports|.
+    1. Return |imports|.
+</div>
+
+<div algorithm>
+    The <dfn method for="Module">customSections(moduleObject, sectionName)</dfn> method, when invoked, performs the following steps:
+    1. Let |bytes| be |moduleObject|.\[[Bytes]].
+    1. Let |customSections| be an empty [=list=] of [=ArrayBuffer=]s.
+    1. For each [=custom section=] |customSection| in the binary format of |bytes|,
+        1. Let |name| be the <code>name</code> of the custom section, [=UTF-8 decoder|decoded as UTF-8=].
```

Comment:
Will this decoding ever fail? If not, an assert might help. If so, specifying what kind of error to throw is important.